configs/nvme-storage-configurator/daemonset.yaml

```diff
@@ -0,0 +1,51 @@
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: nvme-storage-configurator
+  labels:
+    app: nvme-storage-configurator
+  namespace: kube-system
+spec:
+  selector:
+    matchLabels:
+      app: nvme-storage-configurator
+  template:
+    metadata:
+      labels:
+        app: nvme-storage-configurator
+    spec:
+      nodeSelector:
+        node-role.kubernetes.io/konflux-tenants: ""
```

Comment:
Question: IIUC, we will need to label the worker pool with `konflux-tenants` for the disk configuration to happen?

Why not running this on all the non infra workers? the daemon set will configure the nvme disk if there is one otherwise it will do nothing.  Two advantages of running this on all the non infra workers are:

- no need to label the worker pool
- if we ever start to use EC2 instances with nvme disk for the workers where Konflux will run eventually, no change will be required.

Reply:
Lets discuss this.

Reply:
I suggest that we will first run it on a subset of nodes just to see it's stable. I would like also to verify the behavior of the daemonset after the node reboots and after cluster upgrade.

Reply:
@gbenhaim We will run it on a new node pool with the help of NodeSelector. DaemonSet will only run its pod on the nod ematching the node slector label. We won't touch existing worker or infra nodes.